File: android/src/test/java/de/idrinth/habitevaluator/android/ui/EmotionPairAdapterTest.java
package de.idrinth.habitevaluator.android.ui;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.util.ArrayList;
import java.util.List;

import de.idrinth.habitevaluator.shared.model.EmotionPair;

import static org.junit.jupiter.api.Assertions.*;

class EmotionPairAdapterTest {

    private List<EmotionPair> emotionPairs;
    private TestDeleteListener deleteListener;
    private TestClickListener clickListener;
    private EmotionPairAdapter adapter;

    @BeforeEach
    void setUp() {
        emotionPairs = new ArrayList<>();
        deleteListener = new TestDeleteListener();
        clickListener = new TestClickListener();
        adapter = new EmotionPairAdapter(emotionPairs, deleteListener, clickListener);
    }

    @Test
    void testEmptyAdapterItemCount() {
        assertEquals(0, adapter.getItemCount());
    }

    @Test
    void testItemCountWithPairs() {
        emotionPairs.add(createPair("sad", "happy"));
        emotionPairs.add(createPair("anxious", "calm"));
        emotionPairs.add(createPair("tired", "energetic"));
        assertEquals(3, adapter.getItemCount());
    }

    @Test
    void testItemCountAfterAdding() {
        assertEquals(0, adapter.getItemCount());
        emotionPairs.add(createPair("sad", "happy"));
        assertEquals(1, adapter.getItemCount());
    }

    @Test
    void testItemCountAfterRemoving() {
        EmotionPair pair = createPair("sad", "happy");
        emotionPairs.add(pair);
        assertEquals(1, adapter.getItemCount());
        emotionPairs.remove(pair);
        assertEquals(0, adapter.getItemCount());
    }

    @Test
    void testItemCountAfterClearing() {
        emotionPairs.add(createPair("sad", "happy"));
        emotionPairs.add(createPair("anxious", "calm"));
        assertEquals(2, adapter.getItemCount());
        emotionPairs.clear();
        assertEquals(0, adapter.getItemCount());
    }

    @Test
    void testAdapterWithNullListeners() {
        EmotionPairAdapter nullListenersAdapter = new EmotionPairAdapter(emotionPairs, null, null);
        emotionPairs.add(createPair("sad", "happy"));
        assertEquals(1, nullListenersAdapter.getItemCount());
    }

    private EmotionPair createPair(String negative, String positive) {
        EmotionPair pair = new EmotionPair();
        pair.setNegativeLabel(negative);
        pair.setPositiveLabel(positive);
        return pair;
    }

    private static class TestDeleteListener implements EmotionPairAdapter.OnDeleteListener {
        EmotionPair lastDeleted;

        @Override
        public void onDelete(EmotionPair pair) {
            lastDeleted = pair;
        }
    }

    private static class TestClickListener implements EmotionPairAdapter.OnClickListener {
        EmotionPair lastClicked;

        @Override
        public void onClick(EmotionPair pair) {
            lastClicked = pair;
        }
    }
}

File: android/src/test/kotlin/de/idrinth/habitevaluator/android/ui/EmotionPairAdapterTest.kt
package de.idrinth.habitevaluator.android.ui

import org.junit.Test
import org.junit.Assert.*

class EmotionPairAdapterTest {

    @Test
    fun testAdapterReplacedByCompose() {
        // EmotionPairAdapter replaced by LazyColumn in EmotionalStateScreen composable
        assertTrue(true)
    }

    @Test
    fun testEmotionPairListCanBeEmpty() {
        val pairs = mutableListOf<Pair<String, String>>()
        assertEquals(0, pairs.size)
    }

    @Test
    fun testEmotionPairListCanHoldMultipleItems() {
        val pairs = mutableListOf(
            Pair("sad", "happy"),
            Pair("anxious", "calm"),
            Pair("tired", "energetic")
        )
        assertEquals(3, pairs.size)
    }

    @Test
    fun testEmotionPairListCanRemoveItems() {
        val pairs = mutableListOf(Pair("sad", "happy"), Pair("anxious", "calm"))
        pairs.removeAt(0)
        assertEquals(1, pairs.size)
    }

    @Test
    fun testEmotionPairListClearRemovesAll() {
        val pairs = mutableListOf(Pair("sad", "happy"), Pair("anxious", "calm"))
        pairs.clear()
        assertEquals(0, pairs.size)
    }
}
